### PR TITLE
Implement Dynamic Port Mapping for Function Containers for Local Deployment 

### PR DESCRIPTION
**Pull request by Kaleab-git**
Needs Review - @mcopik 

This PR implements port mapping for function containers when SeBS is running local deployment on non-linux pllatforms.

On local start of N deployments
 - Assigns N different free ports to N containers in the range 9000 - 9999
 - Maps each container's `port` to host's `port`
 - Ports that are not available on the host system and ports that belong to containers that are still running are skipped
 - Saves allocated ports to configuration
 - Sets function url to `localhost:port` on non-linux platforms instead of the `DockerIPAddress:port`

### CODE REVIEW

File: sebs/local/function.py [hunk 1]
```diff
@@ -53,17 +54,21 @@ def __init__(
...
-        )
-        if not self._url:
-            self.logging.error(
-                f"Couldn't read the IP address of container from attributes "
-                f"{json.dumps(self._instance.attrs, indent=2)}"
-            )
-            raise RuntimeError(
-                f"Incorrect detection of IP address for container with id {self._instance_id}"
+
+        if is_linux():
```
Comment (mcopik) [inline sebs/local/function.py:58]: In the other PR, you added the if condition to storage and also checked if SeBS is running under WSL. Do we need that here as well?

File: sebs/utils.py [hunk 2]
```diff
@@ -251,6 +252,9 @@ def logging_handlers(self, handlers: LoggingHandlers):
 def has_platform(name: str) -> bool:
     return os.environ.get(f"SEBS_WITH_{name.upper()}", "False").lower() == "true"
 
+# Check if the system is Linux and that it's not WSL
+def is_linux() -> bool:
+    return platform.system() == "Linux" and "microsoft" not in platform.release().lower()
```
Comment (mcopik) [inline sebs/utils.py:257]: Now I see :-) Can we use that in `sebs/storage/minio.py`?

File: sebs/local/config.py [hunk 3]
```diff
@@ -24,14 +26,27 @@ def deserialize(config: dict, cache: Cache, handlers: LoggingHandlers) -> Creden
...
+        self._path: str = ""
         self._storage = storage_cfg
+        self._allocated_ports = set()
 
     @property
     def storage_config(self) -> Optional[MinioConfig]:
         return self._storage
 
+    @property
+    def path(self) -> str:
```
Comment (mcopik) [inline sebs/local/config.py:38]: What is the point of path? I don't see it being used anywhere? I also don't see it being assigned any non-null value?

**Review by mcopik — CHANGES_REQUESTED** [review]

**Comment by mcopik** [comment]
@Kaleab-git Thank you very much for your help - very good work! I left three comments - mostly small changes :)

**Comment by mcopik** [comment]
@Kaleab-git Hi! Do you plan to finish the PR? I'd be happy to merge it and include it in the upcoming release.

**Comment by Kaleab-git** [comment]
> @Kaleab-git Hi! Do you plan to finish the PR? I'd be happy to merge it and include it in the upcoming release.

Hey! Yes, I'll work on it today and address all comments. Sorry for the delay.

File: sebs/utils.py (hunk 2 above)
Comment (Kaleab-git) [reply sebs/utils.py:257]: 👍

File: sebs/local/config.py (hunk 3 above)
Comment (Kaleab-git) [reply sebs/local/config.py:38]: we're using it in `LocalConfig`'s (class in `sebs/local/config.py`) serialize method to store allocated ports.

**Comment by Kaleab-git** [comment]
Hey, I think I've addressed all points. Please me know if there's anything I should change. Thanks!

**Comment by mcopik** [comment]
@Kaleab-git Thank you for the PR :-) I took a while but I was finally able to test it on WSL and verify it works great!

**Pull request merged**